src/test/java/com/amazon/opendistroforelasticsearch/security/support/ModuleInfoTests.java

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright 2015-2019 _floragunn_ GmbH
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * Portions Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.security.support;
+
+import com.amazon.opendistroforelasticsearch.security.auth.HTTPAuthenticator;
+import java.io.IOException;
+import java.util.HashMap;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.junit.Assert;
+import org.junit.Test;
+import org.powermock.api.mockito.PowerMockito;
+
+public class ModuleInfoTests {
+
+    @Test
+    public void testGetAsMap1() {
+        ModuleType moduleType = ModuleType.getByDefaultImplClass(String.class);
+        ModuleInfo moduleInfo = new ModuleInfo(moduleType, "classname");
+
+        HashMap<String, String> hashMap = new HashMap<>();
+        hashMap.put("type", "UNKNOWN");
+        hashMap.put("description", "Unknown type");
+        hashMap.put("is_advanced_module", "true");
+        hashMap.put("default_implementation", null);
+        hashMap.put("actual_implementation", "classname");
+        hashMap.put("version", "");
+        hashMap.put("buildTime", "");
+        hashMap.put("gitsha1", "");
+
+        Assert.assertEquals(hashMap, moduleInfo.getAsMap());
+    }
+
+    @Test
+    public void testGetAsMap2() throws IOException {
```

Comment:
The source method does not declare that it `throws IOException`, nor does the other test method for this method, testGetAsMap1()

Reply:
StreamInput requires it :(